### PR TITLE
Update to nalgebra 0.33.0 and simba 0.9.0

### PR DESCRIPTION
**Pull request by contagon**
nalgebra and simba had a handful of breaking changes in their most recent updates. This should upgrade num-dual to make it compatible. Specifically, the changes were

- Moving of type in `nalebgra::Allocator` generics into a generic associated type on `Allocator::Buffer`. Majority of changes were in this
- simba added `SupersetOf<f32>` as a necessary trait bound for `ComplexField` and `RealField`
- in `SimdValue`, lanes changed from a static fn to a associated const.

All tests seemed to pass, but let me know if there's any other issues. I'll be using it in my own project for the next week or so and will update this if I hit any bugs although I don't expect any.

Also, I'm not sure if the `Allocator` changes would result in any potential simplifying that coule be made... my changes are mostly just to get things working. Let me know if there's any areas that may be worth investigating as far as simplifying things go.

### CODE REVIEW

**Comment by g-bauer** [comment]
That looks great. I successfully ran the tests in FeOs based on this PR. 👍

File: src/derivative.rs [hunk 1]
```diff
@@ -406,18 +406,15 @@ where
 
 impl<T, R: Dim, C: Dim> nalgebra::SimdValue for Derivative<T, T::Element, R, C>
 where
-    DefaultAllocator: Allocator<T, R, C> + Allocator<T::Element, R, C>,
+    DefaultAllocator: Allocator<R, C> + Allocator<R, C>,
```
Comment (prehner) [inline src/derivative.rs:409]:
```suggestion
    DefaultAllocator: Allocator<R, C>,
```

File: src/derivative.rs [hunk 2]
```diff
@@ -541,8 +538,8 @@ impl<TSuper, FSuper, T, F, R: Dim, C: Dim> SubsetOf<Derivative<TSuper, FSuper, R
 where
     TSuper: DualNum<FSuper> + SupersetOf<T>,
     T: DualNum<F>,
-    DefaultAllocator: Allocator<T, R, C>,
-    DefaultAllocator: Allocator<TSuper, R, C>,
+    DefaultAllocator: Allocator<R, C>,
+    DefaultAllocator: Allocator<R, C>,
```
Comment (prehner) [inline src/derivative.rs:542]:
```suggestion
```

**Review by prehner — APPROVED** [review]
Great work, thanks! The changes in nalgebra make the code much more readable. I found two lines where we can get rid of the duplicate trait bounds.

File: src/derivative.rs (hunk 1 above)
Comment (contagon) [reply src/derivative.rs]: Ooh, that's a great catch. Fixed!

File: src/derivative.rs (hunk 2 above)
Comment (contagon) [reply src/derivative.rs]: Fixed!

**Pull request merged**